src/types/fieldExtractor.ts

```diff
@@ -1,13 +1,46 @@
-import { getTrait, NFT } from './nft';
+import { getTrait, NFT, NftFactory } from './nft';
 
-export enum CustomFieldExtractor {
+export type ExtractorTypes = {
+  title?: TitleExtractorTypes
+  id?: IdExtractorTypes
+}
+
+export enum TitleExtractorTypes {
   METADATA_NAME = 'metadata.name',
   ATTRIBUTES_TRAIT_SONG_TITLE = 'attributes.trait.songTitle',
-  ATTRIBUTES_TRAIT_TRACK = 'attributes.trait.track'
+  ATTRIBUTES_TRAIT_TRACK = 'attributes.trait.track',
+}
+
+export enum IdExtractorTypes {
```

Comment:
Not much substance to this just yet, but after looking at all the implementations of `mapNFTtoTrackIDs`, there are a few extractable strategies to deposit here - if we're happy with the approach